c++/src/kj/common.h

```diff
@@ -1940,22 +1940,31 @@ class ArrayPtr: public DisallowConstCopyIfNotConst<T> {
   // Syntax sugar for invoking U::from.
   // Used to chain conversion calls rather than wrap with function.
 
-  inline void fill(T t) { 
+  inline void fill(T t) {
     // Fill the area by copying t over every element.
 
-    for (size_t i = 0; i < size_; i++) { ptr[i] = t; } 
+    for (size_t i = 0; i < size_; i++) { ptr[i] = t; }
     // All modern compilers are smart enough to optimize this loop for simple T's.
     // libc++ std::fill doesn't have memset specialization either.
   }
 
+  inline void fill(ArrayPtr<const T> other) {
+    // Fill the area by copying each element of other, in sequence, over every element.
+    if (other.size() == 0) return;
+    T* __restrict__ dst = begin();
+    for (size_t s = size_, i = 0; i < s; i++) {
+      dst[i] = other[i % other.size()];
```

Comment:
I suppose `fill()` will only overwrite whatever T's assignment operator writes to, so it might leave some old content in elements, particularly for structures that have padding bytes.  Not sure if that actually matters for anything other than security applications, though, or things that compute equality or hashes from raw byte content?